File: Services/Shared/Result.js
class Result {
    constructor(int, data = null, message = null) {
        this.success = Result.mapInt(int);
        this.data = data;
        this.message = message || Result.defaultMessages(int);
    }

    static mapInt(int) {
        if (int === 1) {
            return{
                status: true,
                type: "Success"
            }
        } else if (int === 0) {
            return {
                status: false,
                type: "Failure"
            }
        } else {
            return {
                status: false,
                type: "Error"
            }
        }
    }

    static defaultMessages(int) {
        switch (int) {
            case 1:
                return "Succeeded Executing Transaction.";
            case 0:
                return "Failed Executing Transaction.";
            default:
                return "Error Executing Transaction.";
        }
    }

    static success(data, message = null) {
        return new Result(1, data, message);
    }

    static fail(message = null) {
        return new Result(0, null, message);
    }

    static error(message = null) {
        return new Result(-1, null, message);
    }

    toString() {
        return `${this.success.type} : ${this.message}`;
    }

    toJSON() {
        return {
            success: this.success,
            data: this.data,
            message: this.message
        };
    }
}

module.exports = Result;

File: Services/Shared/Result.py
class Result:
    def __init__(self, int_value, data=None, message=None):
        self.success = self.map_int(int_value)
        self.data = data
        self.message = message or self.default_messages(int_value)

    @staticmethod
    def map_int(int_value):
        if int_value == 1:
            return {"status": True, "type": "Success"}
        elif int_value == 0:
            return {"status": False, "type": "Failure"}
        else:
            return {"status": False, "type": "Error"}

    @staticmethod
    def default_messages(int_value):
        if int_value == 1:
            return "Succeeded Executing Transaction."
        elif int_value == 0:
            return "Failed Executing Transaction."
        else:
            return "Error Executing Transaction."

    @classmethod
    def success(cls, data=None, message=None):
        return cls(1, data, message)

    @classmethod
    def fail(cls, message=None):
        return cls(0, None, message)

    @classmethod
    def error(cls, message=None):
        return cls(-1, None, message)

    def __str__(self):
        return f"{self.success['type']} : {self.message}"

    def to_json(self):
        return {
            "success": self.success,
            "data": self.data,
            "message": self.message
        }
